src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/RuleFactory.cs

```diff
@@ -34,7 +37,8 @@ public IRule[] CreateRules(RuleSettings settings, IRuleRegistrationContext conte
                 new CannotSealType(settings, context),
                 new EnumsMustMatch(settings, context),
                 new MembersMustExist(settings, context),
-                new AttributesMustMatch(settings, context, _excludeAttributesFiles)
+                new AttributesMustMatch(settings, context, _excludeAttributesFiles),
+                new CannotChangeParameterName(settings, context, _enableRuleCannotChangeParameterName),
```

Comment:
Wouldn't it make more sense to not even instantiate the rule if `_enableRuleCannotChangeParameterName` is false here?

Reply:
Depending on the way you approach this, that may require changing the `CreateRules` method's return type to a `List<IRule>`, since we won't know its size ahead of time. Depending on our decision there, I think we should save that for another change.